lib/streamlit/elements/widgets/time_widgets.py

```diff
@@ -73,47 +86,94 @@
 )
 
 
-def _parse_date_value(
-    value: Literal["today", "default_value_today"] | DateValue,
-) -> tuple[list[date] | None, bool]:
-    parsed_dates: list[date]
-    range_value: bool = False
+def _convert_timelike_to_time(value: TimeValue) -> time:
+    if value == "now":
+        # Set value default.
+        return datetime.now().time().replace(second=0, microsecond=0)
+
+    if isinstance(value, str):
+        try:
+            return time.fromisoformat(value)
+        except ValueError:
+            try:
+                return (
+                    datetime.fromisoformat(value)
+                    .time()
+                    .replace(second=0, microsecond=0)
+                )
+            except ValueError:
+                # We throw an error below.
+                pass
+
+    if isinstance(value, datetime):
+        return value.time().replace(second=0, microsecond=0)
+
+    if isinstance(value, time):
+        return value
+
+    raise StreamlitAPIException(
+        "The type of value should be one of datetime, time, ISO string or None"
+    )
+
+
+def _convert_datelike_to_date(
+    value: ExtendedScalarDateValue,
+) -> date:
+    if isinstance(value, datetime):
+        return value.date()
+
+    if isinstance(value, date):
+        return value
+
+    if value in {"today", "default_value_today"}:
+        return datetime.now().date()
+
+    if isinstance(value, str):
+        try:
+            return date.fromisoformat(value)
+        except ValueError:
+            try:
+                return datetime.fromisoformat(value).date()
+            except ValueError:
+                # We throw an error below.
+                pass
+
+    raise StreamlitAPIException(
+        'Date value should either be an date/datetime or an ISO string or "today"'
+    )
+
+
+def _parse_date_value(value: DateValue) -> tuple[list[date] | None, bool]:
     if value is None:
-        return None, range_value
-    if value == "today":
-        parsed_dates = [datetime.now().date()]
-    elif value == "default_value_today":
-        parsed_dates = [datetime.now().date()]
-    elif isinstance(value, datetime):
-        parsed_dates = [value.date()]
-    elif isinstance(value, date):
-        parsed_dates = [value]
-    elif isinstance(value, (list, tuple)):
-        if len(value) not in {0, 1, 2}:
-            raise StreamlitAPIException(
-                "DateInput value should either be an date/datetime or a list/tuple of "
-                "0 - 2 date/datetime values"
-            )
+        return None, False
 
-        parsed_dates = [v.date() if isinstance(v, datetime) else v for v in value]
-        range_value = True
+    value_tuple: Sequence[ExtendedScalarDateValue]
+
+    if isinstance(value, (list, tuple)):
```

Comment:
nit: since `DateValue` uses `Sequence` as the type, I am wondering whether using the check for `Sequence` here like `isinstance(value, Sequence)` would be more accurate